newIDE/app/src/Export/Builds/BuildProgressAndActions.js

```diff
@@ -195,60 +217,75 @@ const BuildProgressAndActions = ({
               )}
             </Line>
             {hasTimedOut && (
-              <ColumnStackLayout>
-                <Line justifyContent="flex-end">
-                  <Text>
+              <Column>
+                <Line justifyContent="flex-end" noMargin>
+                  <Text noMargin>
                     <Trans>Something wrong happened :(</Trans>
                   </Text>
                 </Line>
-                <Line justifyContent="flex-end">
-                  <EmptyMessage style={{ justifyContent: 'flex-end' }}>
+                <Line justifyContent="flex-end" noMargin>
+                  <EmptyMessage
+                    style={{ justifyContent: 'flex-end', padding: 0 }}
+                  >
                     <Trans>
                       It looks like the build has timed out, please try again.
                     </Trans>
                   </EmptyMessage>
                 </Line>
-              </ColumnStackLayout>
+              </Column>
             )}
           </>
         ) : build.status === 'complete' ? (
-          <ColumnStackLayout>
-            <Line expand justifyContent="flex-end">
-              {game && !!build.s3Key && !isBuildPublished && (
-                <RaisedButton
-                  label={<Trans>Publish this build on Liluo.io</Trans>}
-                  onClick={() => onUpdatePublicBuild(build.id)}
-                  disabled={gameUpdating}
-                />
-              )}
-              {game && !!build.s3Key && isBuildPublished && (
-                <FlatButton
-                  label={<Trans>Unpublish this build from Liluo.io</Trans>}
-                  onClick={() => onUpdatePublicBuild(null)}
-                  disabled={gameUpdating}
-                />
-              )}
-              <Spacer />
-              {downloadButtons
-                .filter(button => !!build[button.key])
-                .map(button => (
-                  <React.Fragment key={button.key}>
-                    <RaisedButton
-                      label={i18n._(button.displayName)}
-                      primary
-                      onClick={() => onDownload(button.key)}
+          <ColumnStackLayout noMargin expand>
+            <ResponsiveLineStackLayout
+              expand
+              justifyContent="space-between"
+              noMargin
+            >
+              <ResponsiveLineStackLayout noMargin noColumnMargin>
+                {game && !!build.s3Key && (
+                  <>
+                    <Toggle
+                      label={<Trans>Publish this build on Liluo.io</Trans>}
+                      labelPosition="left"
+                      toggled={isBuildPublished}
+                      onToggle={() => {
+                        onUpdatePublicBuild(isBuildPublished ? null : build.id);
```

Comment:
just to confirm, is this triggering a confirmation modal as well?

Reply:
It doesn't but I think it should indeed! 😅